src/org/exist/xquery/XQueryContext.java

```diff
@@ -2566,24 +2567,42 @@ public LocalVariable markLocalVariables( boolean newContext )
     }
 
 
-    public void popLocalVariables(LocalVariable var) {
-        popLocalVariables(var, null);
+    public void popLocalVariables(final LocalVariable var) {
+        popLocalVariables(var, null, null);
+    }
+
+    public void popLocalVariables(final LocalVariable var, @Nullable final Sequence resultSeq) {
+        popLocalVariables(var, null, resultSeq);
+    }
+
+    public void popLocalVariables(final LocalVariable var, @Nullable final Expression parent) {
+        popLocalVariables(var, parent, null);
     }
 
     /**
      * Restore the local variable stack to the position marked by variable var.
      *
-     * @param  var
-     *
+     * @param var The variable
+     * @param returnExpr the return expression or null
+     * @param resultSeq the result sequence or null
      */
-    public void popLocalVariables(LocalVariable var, Sequence resultSeq)
-    {
+    public void popLocalVariables(final LocalVariable var, @Nullable final Expression returnExpr, @Nullable final Sequence resultSeq) {
```

Comment:
ModuleContext inherits from XQueryContext and delegates popLocalVariables to its parent context. The newly introduced methods should thus be overwritten there as well to delegate to parentContext.

This is causing the failures in the group by test (which have nothing to do with group by itself).